bugwarrior/config/schema.py

```diff
@@ -291,27 +292,35 @@ def validate_config(config: dict, main_section: str, config_path: str) -> dict:
     servicemap = {}
     for target, serviceconfig in configmap.items():
         try:
-            servicemap[target] = serviceconfig['service']
+            servicemap[target] = serviceconfig["service"]
         except KeyError:
             raise_validation_error(
                 f"No option 'service' in section: '{target}'", config_path
             )
 
+    # NOTE: I think this could be less dynamic, maybe by using discriminated unions
+    # https://docs.pydantic.dev/2.0/usage/types/unions/#discriminated-unions-aka-tagged-unions
     # Construct Service Models
     target_schemas = {
-        target: (get_service(service).CONFIG_SCHEMA, ...)
+        target: (
+            Annotated[
+                get_service(service).CONFIG_SCHEMA,
+                BeforeValidator(inject_target(target)),
+            ],
+            ...,
+        )
         for target, service in servicemap.items()
     }
 
     # Construct Validation Model
-    bugwarrior_config_model = pydantic.v1.create_model(
-        'bugwarriorrc',
+    bugwarrior_config_model = pydantic.create_model(
+        "bugwarriorrc",
         __base__=SchemaBase,
-        __validators__={'compute_target': get_target_validator(targets)},
         general=(MainSectionConfig, ...),
-        flavor={
-            flavor: (MainSectionConfig, ...)
-            for flavor in config.get('flavor', {}).values()
+        # NOTE: I add to change this, but not sure about my change though
```

Comment:
Why did you have to change this? Seems like this change will cause a KeyError on `config['flavor']` in `bugwarrior/config/load.py` but evidently we don't have any automated tests for flavors.